sccp/figures/figureThomsonDrop.py

```diff
@@ -0,0 +1,56 @@
+from sccp.figures.common import (subplotLabel, 
+getSetup, openPf2, savePf2, flattenData)
+from sccp.imports.scRNA import ThompsonXA_SCGenesAD, tensorFy
+from sccp.imports.gating import gateThomsonCells
+from parafac2 import parafac2_nd
+import umap
+import numpy as np
+import json
+import seaborn as sns
+import anndata as ad
+from sccp.figures.commonFuncs.plotFactors import (
+    plotFactors,
+)
+from sccp.figures.commonFuncs.plotUMAP import (
+    plotCmpPerCellType,
+    plotCmpUMAP,
+    points
+)
+import pacmap
+
+
+def makeFigure():
+    data = ThompsonXA_SCGenesAD()
+    rank = 30
+    data.obs["cell_type"] = gateThomsonCells()
+
+    ax, f = getSetup((10,7), (1, 1))
+    subplotLabel(ax)
+
+    sampled_data = data.copy()
+    for drug in data.obs.Drugs.unique():
+        for cell_type in data.obs.cell_type.unique():
+            filt = (sampled_data.obs.cell_type != cell_type) | (
+                    sampled_data.obs.Drugs != drug
```

Comment:
Why do you have to loop over drugs, if you build in the drug and cell type filtering as a logical or?

Reply:
This one was a filter to just drop an equal 1% across the data to just pretend if those cells were never in the data in the first place